docs: add note to use dotenv-expand (#10440) (#10798)
vitejs/vite: docs/guide/env-and-mode.md

console.log(import.meta.env.DB_PASSWORD) // undefined
```

> Also vite uses [dotenv-expand](https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow) to expand the functionality of your environments out of the box, you can check the dependency rules to learn how to use it.

Reviewer: ```suggestion
Also vite uses [dotenv-expand](https://github.com/motdotla/dotenv-expand) to expand variables out of the box. To learn more about the syntax, check out [their docs](https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow).
```

> Also vite uses [dotenv-expand](https://github.com/motdotla/dotenv-expand) to expand variables out of the box. To learn more about the syntax, check out [their docs](https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow).
> 
> Note that if you want to use the `$` inside your environment value, you have to escape it with `\`.

Reviewer 2: ```suggestion
Also, Vite uses [dotenv-expand](https://github.com/motdotla/dotenv-expand) to expand variables out of the box. To learn more about the syntax, check out [their docs](https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow).

Note that if you want to use `$` inside your environment value, you have to escape it with `\`.
```
